app/components/page-components/Chatbot/shared/ChatbotPermissions.vue.ts

```diff
@@ -0,0 +1,129 @@
+import { Component, Prop, Watch } from 'vue-property-decorator';
+import ChatbotBase from 'components/page-components/Chatbot/ChatbotBase.vue';
+import { filter, map, cloneDeep } from 'lodash';
+import { IPermissionCheckboxes } from 'services/chatbot';
+
+@Component({})
+export default class ChatbotAliases extends ChatbotBase {
+  @Prop()
+  value: number;
+
+  paneMenuOpen: boolean = false;
+  text: string = 'Everyone';
+
+  checkboxes: IPermissionCheckboxes = {
+    regular: { checked: true, string: 'Regulars' },
+    subscriber: { checked: true, string: 'Subscribers' },
+    moderator: { checked: true, string: 'Moderators' },
+    streamer: { checked: true, string: 'Streamer' },
+  };
+
+  previousCheckBoxes: IPermissionCheckboxes = {};
+
+  mounted() {
+    this.updateChecks();
+    console.log(this.value);
+  }
+
+  @Watch('value')
+  valueWatcher(newValue: number) {
+    this.updateChecks();
+  }
+
+  @Watch('checkboxes', { deep: true })
+  checkBoxWatcher(newValue: IPermissionCheckboxes) {
+    let value = 0;
+
+    const itemsChecked = filter(newValue, type => type.checked).length;
+
+    if (itemsChecked === 0) {
+      this.$nextTick(() => {
+        this.checkboxes.regular.checked = this.previousCheckBoxes.regular.checked;
+        this.checkboxes.moderator.checked = this.previousCheckBoxes.moderator.checked;
+        this.checkboxes.subscriber.checked = this.previousCheckBoxes.subscriber.checked;
+        this.checkboxes.streamer.checked = this.previousCheckBoxes.streamer.checked;
+      });
+
+      return;
+    }
+
+    this.previousCheckBoxes = cloneDeep(newValue);
+
+    value += newValue.subscriber.checked ? 2 : 0;
+    value += newValue.regular.checked ? 4 : 0;
+    value += newValue.moderator.checked ? 32 : 0;
+    value += newValue.streamer.checked ? 64 : 0;
```

Comment:
There's an enum defined later in this PR.  Why isn't it being used?

Reply:
Done